src/osd/osd_types.h

```diff
@@ -2447,7 +2447,7 @@ struct store_statfs_t
 
   void add(const store_statfs_t& o) {
     total += o.total;
-    available += o.available;
+    available = std::min<uint64_t>(available + o.available, total);
```

Comment:
before we do this, how reproducible is this https://tracker.ceph.com/issues/39634#note-6?

Reply:
I think it should be 1%~3%. And I have seen this at least 4 times, such as in https://tracker.ceph.com/issues/45434, there also has the same issue except that I have fixed.

This qa test will try to fill the cephfs data pool full first and then remove some files to wait for the not full flag being cleared, but sometimes it will fail like this.

I checked the memstore and some others, they all have do this to make sure the `available` won't exceed the `total` in all cases.

Reply:
Honestly I don't like this approach for fixing the issue - it just hides the real bug - stats should be consistent and avail should never go above total - and makes it even harder to find.  But I realize that we might want to have a quick workaround for our QA.
Given the above I would suggest to comment the patched code with ф clear statement that this is a workaround and general fix should be different, log (at a very low debug level, e.g. dout(1)) when this issue happens and create (and leave it open) a proper different bluestore ticket referencing this PR/ticket.

Reply:
how about logging to cluster log when we hit this condition, that way teuthology tests will fail and we can go dig deeper?

Reply:
@lxbsz - so you're saying so far this statfs discrepancy has been observed for memstore only, right? At least you're unaware of similar cases for non-memstore runs, aren't you?

Reply:
Yeah, right. Because our qa `test_full` test cases are only using the memstore. But I have checked the memstore code, it has already make sure the the `available` won't exceed the `total` in any case.